full-moon/src/ast/visitors.rs

```diff
@@ -412,24 +437,35 @@ impl VisitMut for LocalAssignment {
             type_specifiers = std::iter::repeat::<Option<Self>>(None);
         }
 
+        let mut new_attributes = Vec::new();
         let mut new_type_specifiers = Vec::new();
         let mut new_names = Punctuated::new();
 
         for parameter_pair in self.name_list.into_pairs() {
             let parameter_tuple = parameter_pair.into_tuple();
             let parameter = parameter_tuple.0.visit_mut(visitor);
+            let attribute = attributes
+                .next()
+                .and_then(|attribute| attribute)
```

Comment:
Why do we do this? Can we just `.flatten()` instead?

Reply:
Not sure. Changed